drivers/dai/intel/dmic/dmic.c

```diff
@@ -743,7 +743,9 @@ static const struct dai_config *dai_dmic_get_config(const struct device *dev, en
 {
 	struct dai_intel_dmic *dmic = (struct dai_intel_dmic *)dev->data;
 
-	__ASSERT_NO_MSG(dir == DAI_DIR_RX);
+	if (dir != DAI_DIR_RX)
```

Comment:
Are you sure the exit value is checked everywhere?
i.e.

dai-zephyr.c, line 159:

![image](https://user-images.githubusercontent.com/84508161/210207415-8afcf651-49f9-4893-b7cc-274957a294fa.png)

dereferenced without checking for null 

same in dai.c line 157:
![image](https://user-images.githubusercontent.com/84508161/210207447-37392fe6-a10c-46db-94d6-0cf42d5a607a.png)

Please add checking for null at every call to the function

Reply:
@marcinszkudlinski Ack, let me check. I was only fixing cases where the assert is already hit now (meaning you cannot actually enable asserts in builds to catch other errors). But true, need to check all cases.

Reply:
@marcinszkudlinski both cases now handled https://github.com/thesofproject/sof/pull/6897 and https://github.com/thesofproject/sof/pull/6899 

Given a) there are no Zephyr in-tree inproper usage and b) asserts are not enable by default (yet) in SOF builds, I'd propose we proceed and merge this fix.